Ensure the device is correctly unregistered after logout in all scenarios

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #13611 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
This PR fixes the linked issue by ensuring we unregister the device with our servers for each logout, we achieve this by moving the unregistration logic to the repository, which will ensure it will be called in all scenarios.

Failing to unregister the device was causing issues when the user has to change the account during the login that would prevent registering using the new account, this was because the app will believe the device is already registered.

### Steps to reproduce
1. Use two accounts: account A and account B, and a site connected only to Account A.
1. Start login and enter the site address.
1. Continue using account B.
1. In the Account Mismatch screen, tap on "Login using a different account"
1. Now enter the site address, and use account A.
1. Test push notifications.

### Testing information
- Confirm push notifications are received.
- Open the MC push notifications dashboard and confirm your device is registered with Account A.

### The tests that have been performed
^

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Ensure that the AccountRepository#logout will unregister the device from push notifications

hichamboushaba committed:
Simplify logout code in the settings

hichamboushaba committed:
Fix unit tests

hichamboushaba committed:
Add a unit test

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsPresenter.kt:
```diff
@@ -21,37 +15,27 @@ class AppSettingsPresenter @Inject constructor(
         coroutineScope.launch {
             accountRepository.logout().let {
                 if (it) {
+                    clearCardReaderDataAction()
```
This is a change from before: we won't clear the card reader data until the logout is successful. I feel this is a better approach, let me know what you think.

irfano replied:
Before, we were not clearing card reader data after logout, but now we will. This was a missing feature, and you have added it, correct?

hichamboushaba replied:
Not exactly, we were clearing it, but it was [done](https://github.com/woocommerce/woocommerce-android/blob/819eea50ea769832c9e26e9b6880c4ebe78427d5/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsPresenter.kt#L34) before the logout, which means that if for some reason the logout failed, the data was still cleared.

irfano reviewed (approved):
LGTM! 👍🏻 I confirm that it was not working before and has been fixed with this PR.

hichamboushaba committed:
Ignore device unregistration errors and continue logout

This aligns with the behavior we had in AppSettingsPresenter and is needed to fix UI tests

hichamboushaba commented:
@JorgeMucientes a heads up, we shipped this fix to the release branch, please create a new beta to include it at your earliest convenience.
